datastore/sample.rb

```diff
@@ -0,0 +1,748 @@
+# Copyright 2016 Google, Inc
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require "gcloud"
+
+def incomplete_key
+  gcloud = Gcloud.new
+  datastore = gcloud.datastore
+
+  # [START incomplete_key]
+  task_key = datastore.key "Task"
+  # [END incomplete_key]
+end
+
+def named_key
+  gcloud = Gcloud.new
+  datastore = gcloud.datastore
+
+  # [START named_key]
+  task_key = datastore.key "Task", "sampleTask"
+  # [END named_key]
+end
+
+def key_with_parent
+  gcloud = Gcloud.new
+  datastore = gcloud.datastore
+
+  # [START key_with_parent]
+  task_key = datastore.key "Task", "sampleTask"
+  task_key.parent = datastore.key "TaskList", "default"
+  # [END key_with_parent]
+
+  task_key
+end
+
+def key_with_multilevel_parent
+  gcloud = Gcloud.new
+  datastore = gcloud.datastore
+
+  # [START key_with_multilevel_parent]
+  user_key = datastore.key "User", "alice"
```

Comment:
This doesn't exist right now, but we could add the option so that this code would also work:

```ruby
user_key = datastore.key "User", "alice"
task_list_key = datastore.key "TaskList", "default", parent: user_key
task_key = datastore.key "Task", "sampleTask", parent: task_list_key
task_key
```

@remi, is there some other approach that you think would be simpler/easier to understand?